### DISCUSSION TITLE
Is Rollup plugin errors `column` property zero or one-based?

### DISCUSSION THREAD

**Discussion by ericmorand** [rollup/rollup#5279] [Q&A]
It is not clear from the documentation if the `column` property of the `pos` property of errors emitted through `this.error` is zero or one-based.

Do anyone have the definitive answer?

https://rollupjs.org/plugin-development/#this-error

**Comment by lukastaegert** [comment]
`pos` is zero-based, `loc.column` is also zero-based, and `loc.line` is one-based. You can easily see this when generating an error in the Rollup REPL and opening the browser developer tools as the error is logged and you can inspect its properties.

https://rollupjs.org/repl/?version=4.6.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMiUzQyUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0F0cnVlJTdEJTdE

Rollup inherited this from `acorn`.

**Comment by ericmorand** [comment]
Clear. It explains why my plugin errors position are shifted one character to the right when displayed by rollup: Twing's errors position are one-based, both line and column. I'll adjust. 

Thanks a lot.

**Comment by ericmorand** [comment]
@lukastaegert actually, I already fixed it earlier today, I forgot about that. And the errors look perfect now:

https://github.com/rollup/rollup/discussions/5280

Anyway, thanks for the confirmation that pos / column is zero-based.
